crates/block-buffer-manager/src/block_buffer_manager.rs

```diff
@@ -180,6 +182,7 @@ impl BlockBufferManager {
         info!("init block_buffer_manager with latest_commit_block_number: {:?} block_number_to_block_id: {:?}", latest_commit_block_number, block_number_to_block_id);
         let mut block_state_machine = self.block_state_machine.lock().await;
         // When init, the latest_finalized_block_number is the same as latest_commit_block_number
+        block_state_machine.next_executed_block_number = latest_commit_block_number + 1;
```

Comment:
I doubt if we can assume this assumption. Since both consensus layer and execution layer is fully pipelined. Can we assume the gap always be one?